common/files/files.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package files
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+// FileScanner is a struct which reads the content of given file path and allows
+// to read it line by line. Wrapping into a struct simplifies closing the file
+// descriptor.
+type FileScanner struct {
```

Comment:
Err, why did this need a new class? In the only place we use it, we defer closing the fd within the same function. All the other uses call the wrappers here for the scanner object. Was there some other motivation behind this?
If there isn't any other reason, I would prefer to simply have something like this in `GetMountInfo`
```golang
	filepath := "/proc/self/mountinfo"
	fMounts, err := os.Open(filepath)
	if err != nil {
		return nil, fmt.Errorf("failed to open mount information at %s: %s", filepath, err)
	}
	defer fMounts.Close()

	scanner := bufio.NewScanner(fMounts)
	scanner.Split(bufio.ScanLines)
```

Reply:
Good point. In my previous version of PR, I was using that FileScanner in several places (because I was parsing `/proc/mounts` multiple times in slightly different ways), but now with `mountinfo` module I'm using it only once.

Reply:
cool. thanks!